istio-control/istio-discovery/templates/clusterrole.yaml

```diff
@@ -48,3 +48,100 @@ rules:
   verbs: ["get", "list", "watch"]
 ---
 {{ end }}
+
+{{ if .Values.global.istiod.enabled }}
+# Dedicated cluster role - istiod will use fewer dangerous permissions ( secret access in particular ).
```

Comment:
nit: this comment could probably use some love, since the context will quickly get lost going forward. Might want to expand on what you mean by "dangerous permissions" and also answer the question "fewer than what?".